systems/controllers/constructQPDataPointerMex.cpp

```diff
@@ -0,0 +1,329 @@
+  const mxArray *pobj;
+
+  pobj = myGetField(myGetField(rpc_obj, "position_indices"), "r_leg_kny");
+  Map<VectorXd>r_leg_kny(mxGetPr(pobj), mxGetNumberOfElements(pobj));
+  rpc->position_indices.r_leg_kny = r_leg_kny.cast<int>();
+
+  pobj = myGetField(myGetField(rpc_obj, "position_indices"), "l_leg_kny");
+  Map<VectorXd>l_leg_kny(mxGetPr(pobj), mxGetNumberOfElements(pobj));
+  rpc->position_indices.l_leg_kny = l_leg_kny.cast<int>();
+
+  pobj = myGetField(myGetField(rpc_obj, "position_indices"), "r_leg");
+  Map<VectorXd>r_leg(mxGetPr(pobj), mxGetNumberOfElements(pobj));
+  rpc->position_indices.r_leg = r_leg.cast<int>();
+
+  pobj = myGetField(myGetField(rpc_obj, "position_indices"), "l_leg");
+  Map<VectorXd>l_leg(mxGetPr(pobj), mxGetNumberOfElements(pobj));
+  rpc->position_indices.l_leg = l_leg.cast<int>();
+
+  pobj = myGetField(myGetField(rpc_obj, "position_indices"), "r_leg_ak");
+  Map<VectorXd>r_leg_ak(mxGetPr(pobj), mxGetNumberOfElements(pobj));
+  rpc->position_indices.r_leg_ak = r_leg_ak.cast<int>();
+
+  pobj = myGetField(myGetField(rpc_obj, "position_indices"), "l_leg_ak");
+  Map<VectorXd>l_leg_ak(mxGetPr(pobj), mxGetNumberOfElements(pobj));
+  rpc->position_indices.l_leg_ak = l_leg_ak.cast<int>();
+
+  rpc->body_ids.r_foot = (int) mxGetScalar(myGetField(myGetField(rpc_obj, "body_ids"), "r_foot"));
+  rpc->body_ids.l_foot = (int) mxGetScalar(myGetField(myGetField(rpc_obj, "body_ids"), "l_foot"));
+  rpc->body_ids.pelvis = (int) mxGetScalar(myGetField(myGetField(rpc_obj, "body_ids"), "pelvis"));
+
+  pobj = myGetField(rpc_obj, "actuated_indices");
+  Map<VectorXd>actuated_indices(mxGetPr(pobj), mxGetNumberOfElements(pobj));
+  rpc->actuated_indices = actuated_indices.cast<int>();
+
+  return;
+}
+
+void mexFunction(int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[])
+{
+  if (nrhs<1) mexErrMsgTxt("usage: ptr = constructQPDataPointerMex(robot_obj, params_sets, robot_property_cache, B, umin, umax, terrain_map_ptr, gurobi_opts);");
+
+  if (nrhs == 1) {
+    // By convention, calling the constructor with just one argument (the pointer) should delete the pointer
+    if (isa(prhs[0],"DrakeMexPointer")) { 
+      destroyDrakeMexPointer<NewQPControllerData*>(prhs[0]);
+      return;
+    } else {
+      mexErrMsgIdAndTxt("Drake:constructQPDataPointerMex:BadInputs", "Expected a DrakeMexPointer (or a subclass)");
+    }
+  }
+
+
+  if (nlhs<1) mexErrMsgTxt("take at least one output... please.");
+
+  struct NewQPControllerData* pdata;
+  pdata = new struct NewQPControllerData;
+
+  int narg = 0;
+  
+  // robot_obj
+  pdata->r = (RigidBodyManipulator*) getDrakeMexPointer(prhs[narg]);
+  narg++;
+
+  // param_sets
+  parseAtlasParamSets(prhs[narg], pdata->r, &(pdata->param_sets));
+  narg++;
+
+  // robot_property_cache
+  parseRobotPropertyCache(prhs[narg], &(pdata->rpc));
+  narg++;
+
+  // B
+  pdata->B.resize(mxGetM(prhs[narg]),mxGetN(prhs[narg]));
```

Comment:
B is also in RBM now.  (though I guess it's not parsed in constructModelmex quite yet, see Drake #820)